<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QJackTrip</class>
 <widget class="QMainWindow" name="QJackTrip">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>403</width>
    <height>555</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QJackTrip</string>
  </property>
  <property name="windowIcon">
   <iconset resource="qjacktrip.qrc">
    <normaloff>:/qjacktrip/icon.png</normaloff>:/qjacktrip/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="5" column="0" colspan="2">
     <layout class="QHBoxLayout" name="buttonLayout">
      <item>
       <widget class="QPushButton" name="connectButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="disconnectButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Disconnect</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="exitButton">
        <property name="text">
         <string>Exit</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="addressLabel">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="text">
       <string>&amp;Address of server</string>
      </property>
      <property name="buddy">
       <cstring>addressEdit</cstring>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="typeLabel">
      <property name="text">
       <string>&amp;Run JackTrip as</string>
      </property>
      <property name="buddy">
       <cstring>typeComboBox</cstring>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QComboBox" name="typeComboBox">
      <property name="toolTip">
       <string/>
      </property>
      <item>
       <property name="text">
        <string>Client</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Server</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Hub Client</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Hub Server</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLabel" name="ipLabel">
      <property name="text">
       <string>Looking up external IP address...</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLineEdit" name="addressEdit"/>
    </item>
    <item row="4" column="0" colspan="2">
     <widget class="QTabWidget" name="optionsTabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>391</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <property name="movable">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="basicTab">
       <attribute name="title">
        <string>Basic options</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="3" column="0" colspan="2">
         <widget class="QCheckBox" name="zeroCheckBox">
          <property name="text">
           <string>Set buffer to &amp;zero when underrun occurs</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="autoPatchComboBox">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <item>
           <property name="text">
            <string>Server to clients</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Client loopback</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Client fan out/in but no loopback</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Full Mix</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>No auto patching</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="4" column="0">
         <spacer name="basicVerticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="autoPatchLabel">
          <property name="maximumSize">
           <size>
            <width>130</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Hub auto&amp;patch mode</string>
          </property>
          <property name="buddy">
           <cstring>autoPatchComboBox</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="channelLabel">
          <property name="maximumSize">
           <size>
            <width>130</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>&amp;Number of channels</string>
          </property>
          <property name="buddy">
           <cstring>channelSpinBox</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="channelSpinBox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="value">
           <number>2</number>
          </property>
         </widget>
        </item>
        <item row="6" column="0" colspan="2">
         <layout class="QHBoxLayout" name="aboutLayout">
          <item>
           <spacer name="aboutSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="aboutButton">
            <property name="text">
             <string>About</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="advancedTab">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>Advanced options</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="5" column="0">
         <widget class="QLabel" name="resolutionLabel">
          <property name="text">
           <string>Audio &amp;Bit Resolution</string>
          </property>
          <property name="buddy">
           <cstring>resolutionComboBox</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="clientNameEdit">
          <property name="maxLength">
           <number>64</number>
          </property>
          <property name="placeholderText">
           <string>JackTrip</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="queueLengthSpinBox">
          <property name="minimum">
           <number>2</number>
          </property>
          <property name="value">
           <number>4</number>
          </property>
         </widget>
        </item>
        <item row="6" column="0" colspan="2">
         <layout class="QHBoxLayout" name="connectAudioLayout">
          <item>
           <widget class="QCheckBox" name="connectAudioCheckBox">
            <property name="text">
             <string>&amp;Connect default audio ports</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="queueLengthLabel">
          <property name="text">
           <string>&amp;Queue Buffer Length</string>
          </property>
          <property name="buddy">
           <cstring>queueLengthSpinBox</cstring>
          </property>
         </widget>
        </item>
        <item row="8" column="0" colspan="2">
         <layout class="QHBoxLayout" name="useDefaultsLayout">
          <item>
           <spacer name="useDefaultsSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="useDefaultsButton">
            <property name="text">
             <string>Use &amp;Defaults</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="5" column="1">
         <widget class="QComboBox" name="resolutionComboBox">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <item>
           <property name="text">
            <string>8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>16</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>24</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>32</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="clientNameLabel">
          <property name="text">
           <string>Custom Client &amp;Name</string>
          </property>
          <property name="buddy">
           <cstring>clientNameEdit</cstring>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QSpinBox" name="redundancySpinBox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <spacer name="advancedVerticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="portOffsetLabel">
          <property name="text">
           <string>Port &amp;Offset</string>
          </property>
          <property name="buddy">
           <cstring>portOffsetSpinBox</cstring>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="portOffsetSpinBox">
          <property name="maximum">
           <number>999</number>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="redundancyLabel">
          <property name="text">
           <string>&amp;Redundancy</string>
          </property>
          <property name="buddy">
           <cstring>redundancySpinBox</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="remoteNameEdit">
          <property name="maxLength">
           <number>64</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="remoteNameLabel">
          <property name="text">
           <string>Remote Client Name</string>
          </property>
          <property name="buddy">
           <cstring>remoteNameEdit</cstring>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>403</width>
     <height>30</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <tabstops>
  <tabstop>typeComboBox</tabstop>
  <tabstop>addressEdit</tabstop>
  <tabstop>connectButton</tabstop>
  <tabstop>disconnectButton</tabstop>
  <tabstop>exitButton</tabstop>
  <tabstop>optionsTabWidget</tabstop>
  <tabstop>channelSpinBox</tabstop>
  <tabstop>autoPatchComboBox</tabstop>
  <tabstop>zeroCheckBox</tabstop>
  <tabstop>aboutButton</tabstop>
  <tabstop>clientNameEdit</tabstop>
  <tabstop>remoteNameEdit</tabstop>
  <tabstop>portOffsetSpinBox</tabstop>
  <tabstop>queueLengthSpinBox</tabstop>
  <tabstop>redundancySpinBox</tabstop>
  <tabstop>resolutionComboBox</tabstop>
  <tabstop>connectAudioCheckBox</tabstop>
  <tabstop>useDefaultsButton</tabstop>
 </tabstops>
 <resources>
  <include location="qjacktrip.qrc"/>
 </resources>
 <connections/>
</ui>
